File: node/hello-couchbase.js
var couchbase = require('couchbase');
var cluster = new couchbase.Cluster('couchbase://192.168.4.44');
var bucket = cluster.openBucket('NYC-bucket');

bucket.upsert('testdoc', {name:'Frank'}, function(err, result) {
  if (err) throw err;

  bucket.get('testdoc', function(err, result) {
    if (err) throw err;

    console.log(result.value);
    // {name: Frank}
  });
});
process.exit(0);

File: python/hello-couchbase.py
from couchbase.bucket import Bucket
from couchbase.n1ql import N1QLQuery
from couchbase.exceptions import HTTPError

cb = Bucket('couchbase://192.168.4.44/default')
cb.upsert('u:king_arthur', {'name': 'Arthur',
           'email': 'kingarthur@couchbase.com',
           'interests': ['Holy Grail', 'African Swallows']})
cb.get('u:king_arthur').value
try:
    cb.n1ql_query('CREATE PRIMARY INDEX ON default').execute()
except HTTPError as e:  # Python >2.5
    if e.rc == 59 :
       pass
    else:
        raise
#   print str(e.rc)
#    pass
row_iter = cb.n1ql_query(N1QLQuery('SELECT name FROM default WHERE ' +\
 '$1 IN interests', 'African Swallows'))
for row in row_iter: print row
